Sources/Datadog/Core/Upload/DataUploader.swift

```diff
@@ -41,7 +33,9 @@ internal class UploadURLProvider {
 
     var url: URL {
         var urlComponents = URLComponents(url: urlWithClientToken, resolvingAgainstBaseURL: false)
-        urlComponents?.percentEncodedQueryItems = queryItemProviders.map { $0.value() }
+        if !queryItemProviders.isEmpty {
+            urlComponents?.percentEncodedQueryItems = queryItemProviders.map { $0.value() }
```

Comment:
```suggestion
            urlComponents?.queryItems = queryItemProviders.map { $0.value() }
```
this was fixed here #317

Reply:
👍 , rebased.